src/main/scala/org/scalajs/cli/Scalajsld.scala

```diff
@@ -66,6 +66,27 @@ object Scalajsld {
     }
   }
 
+  private implicit object OutputFileRead extends scopt.Read[OutputFile] {
+    val arity: Int = 1
+    val reads = { (s: String) =>
+      val file = implicitly[scopt.Read[File]].reads(s)
+      if(file.isDirectory){
```

Comment:
I'm not sure this is very user friendly. Essentially, if the output directory doesn't exist, instead of failing or creating it, it falls back to the old behavior (potentially creating a file where a directory is expected).

I think another flag that is mutually exclusive with this one is needed. Unfortunately, the obvious candidate `-d` (same option as in `scalac` / `javac`) is already taken. Not sure what a good flag would be.